### PR TITLE
Add API to add/remove asset locations for an asset

### PR DESCRIPTION
**Pull request by JohnTortugo**
Closes: https://github.com/dotnet/arcade/issues/1796

The publishing pipeline will be able to add/remove asset locations for an asset.

### CODE REVIEW

File: src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/AssetsController.cs [hunk 1]
```diff
@@ -78,5 +80,68 @@ public async Task<IActionResult> GetAsset(int id)
...
+                existing.Type == assetLocation.Type))
+            {
+                return StatusCode((int)HttpStatusCode.NotModified);
+            }
+
+            asset.Locations = asset.Locations ?? new List<Data.Models.AssetLocation>();
+            asset.Locations.Add(assetLocation);
+
+            await _context.SaveChangesAsync();
+            return StatusCode((int)HttpStatusCode.Created);
```
Comment (alexperovich) [inline src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/AssetsController.cs]: This should probably return the AssetLocation object that was added.

File: src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/AssetsController.cs [hunk 2]
```diff
@@ -78,5 +80,68 @@ public async Task<IActionResult> GetAsset(int id)
 
             return Ok(new Asset(asset));
         }
+
+        [HttpPost("{assetId}/locations")]
+        [SwaggerResponse((int)HttpStatusCode.Created, Type = typeof(Channel))]
```
Comment (alexperovich) [inline src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/AssetsController.cs]: This needs to match the status code and type of the return value.

Comment (JohnTortugo) [reply src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/AssetsController.cs]: will fix.

File: src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/AssetsController.cs (hunk 1 above)
Comment (JohnTortugo) [reply src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/AssetsController.cs]: will fix.

**Comment by JohnTortugo** [comment]
Anything else folks?

**Comment by JohnTortugo** [comment]
Ping

**Pull request merged**